ENIGMAsystem/SHELL/Platforms/General/PFmain.cpp

```diff
@@ -166,6 +170,20 @@ int updateTimer() {
   return 0;
 }
 
+void fireEventsFromQueue() {
+  for (unsigned i = 0; i < posted_async_events.size(); i++) {
+    enigma_user::ds_map_clear(enigma_user::async_load);
+
+    std::map<std::string, variant> event = posted_async_events.front();
+
+    posted_async_events.pop();
+
+    for (auto& [key, value] : event) {
+      enigma_user::ds_map_add(enigma_user::async_load, key, value);
+    }
+  }
```

Comment:
This needs reworked because pop decreases the size while is incrementing, you'll miss half the items.

I'd just do it like this:
```cpp
while (!posted_async_events.empty()) {
  
}
```

Reply:
You also need to fire the user event still after filling the map.

Take a look at a couple examples of `instance_event_iterator` which you can just call `myevent_asyncsteam();` on:
https://github.com/enigma-dev/enigma-dev/blob/826b84c599aa3289b4d17533a17de1d14c625335/ENIGMAsystem/SHELL/Platforms/Win32/WINDOWScallback.cpp#L73

Reply:
@RobertBColton 

done!! but there's a problem, where do you want me to add the ```myevent_asyncsteam();``` function?

Reply:
This was resolved, you implemented the event in the component interface struct.